returnn/tf/network.py

```diff
@@ -949,14 +949,18 @@ def _create_layer(self, name, layer_class, **layer_desc):
           print("layer %s/%r output: %r" % (self.name, name, output_template))
         output_template.sanity_check(ignore_placeholder=True)  # placeholder might be overwritten later
         output_template_special_axes = output_template.get_special_axes_dict()
+        placeholder_is_available = True
         if not output_template.available_for_inference and not self.eval_flag:
           from returnn.tf.layers.base import DataNotAvailableLayer
-          return DataNotAvailableLayer(
+          placeholder_is_available = False
+          layer = DataNotAvailableLayer(
             name=layer_desc['name'], network=layer_desc['network'], output=output_template,
-            layer_class=layer_class, layer_desc=layer_desc)
-        layer = layer_class(**layer_desc)
+            layer_class=layer_class, layer_desc=layer_desc,
+            register_as_extern_data=layer_desc.get('register_as_extern_data'))
+        else:
+          layer = layer_class(**layer_desc)
         layer.post_init(layer_desc)
-        layer.output.sanity_check()
+        layer.output.sanity_check(ignore_placeholder=not placeholder_is_available)
```

Comment:
Why is this needed? If it is not set, `sanity_check` will also work.

Reply:
Oh, didn't notice.